workers/docker-worker/src/lib/upload_to_s3.js

```diff
@@ -40,12 +42,21 @@ module.exports = async function uploadToS3 (
       await tmp.writeFile(source);
     } else {
       let stream = fs.createWriteStream(tmp.path);
-      await pipe(source, stream);
+      // TODO: Do we care if this is larger than the original?
+      if (compress) {
+        let gzip = zlib.createGzip();
+        debug(`compressing to ${artifactName} to ${tmp.path}`);
+        await pipe(source, gzip, stream);
+        debug(`compressed ${gzip.bytesWritten} bytes`);
+      } else {
+        await pipe(source, stream);
+      }
     }
-    debug(`wrote source file to ${tmp.path} for ${artifactName}`);
-
     let stat = await fs.stat(tmp.path);
     size = stat.size;
+    httpsHeaders['content-length'] = size;
```

Comment:
Is this safe to do? Can we stop setting content-length in the callers?

Reply:
Yes, it's safe.